File: Java/Clase9/Clase9/src/domain/Cliente.java

package domain;
import java.util.Date;

public class Cliente extends Persona {
    private int idCliente;
    private static int contadorCliente;
    private Date fechaRegistro;
    private boolean vip;
    
    public Cliente (String nombre , Date fechaRegistro , boolean vip){
        super(nombre);
        this.idCliente = ++contadorCliente;
        this.fechaRegistro = fechaRegistro;
        this.vip=vip;
    }

    public Date getFechaRegistro() {
        return this.fechaRegistro;
    }

    public void setFechaRegistro(Date fechaRegistro) {
        this.fechaRegistro = fechaRegistro;
    }

    public boolean isVip() {
        return vip;
    }

    public void setVip(boolean vip) {
        this.vip = vip;
    }

    public int getIdCliente() {
        return this.idCliente;
    }

    public void setIdCliente(int idCliente) {
        this.idCliente = idCliente;
    }

}

File: JavaScript/Clase9/Cliente.js
class Cliente extends Persona{
    static contadorClientes = 0;

    constructor(nombre, apellido, edad, fechaRegistro){
        super(nombre, apellido, edad);
        this._idCliente = ++Cliente.contadorClientes;
        this._fechaRegistro = fechaRegistro;
    }

    get idcliente(){
        return this._idCliente;
    }

    get fechaRegsitro(){
        return this._fechaRegistro;
    }

    set fechaRegistro(fechaRegistro){
        this._fechaRegistro = fechaRegistro;
    }

    toString(){
        return ` 
        ${super.toString()}
        ${this._idCliente}
        ${this._fechaRegistro}`;
    }
}